### PR TITLE
Handle blob from 2.0.0 when stripping legacy _attachments

### PR DESCRIPTION
**Pull request by pasin**
* In `_attachments`, 2.0 blob key will be in `blob_<number>` format, while 2.1 blob key will be in `blob_/<property-pointer>` format.

* In LegacyAttachments.encodeStrippingOldMetaProperties(root, key), detect 2.0 blob key by checking whether the suffix after `blob_` is a digit or not. If 2.0 blob key is detected, remove the attachment and continue.

* Made c4doc_encodeStrippingOldMetaProperties(doc, sk, outError) returns out error.

* When handle incoming rev, check the error when calling c4doc_encodeStrippingOldMetaProperties() and return 500 error code if there is an error occurs. Similar thing was done in RESTListener+Handlers.cc

* Updated c4DocumentTest

#565

### CODE REVIEW

File: C/tests/c4DocumentTest.cc [hunk 1]
```diff
@@ -710,22 +710,20 @@ N_WAY_TEST_CASE_METHOD(C4Test, "Document Legacy Properties 3", "[Database][C]")
...
+    alloc_slice stripped = c4doc_encodeStrippingOldMetaProperties(dict, sk, NULL);
     CHECK(fleece2json(stripped, sk) == "{_attachments:{oldie:{digest:\"sha1-xVVVVVVVVVVVVVVVVVVVVVVVVVU=\"}},foo:[0,{\"@type\":\"blob\",digest:\"sha1-VVVVVVVVVVVVVVVVVVVVVVVVVVU=\"}]}");
 }
 
 
 N_WAY_TEST_CASE_METHOD(C4Test, "Document Legacy Properties 4", "[Database][C]") {
-    // Check that a translated attachment whose digest is different than its blob (i.e. the
-    // attachment was probably modified by a non-blob-aware system) has its digest transferred to
-    // the blob before being deleted. See #507. (Also, the _attachments property should be deleted.)
+    // Check that the 2.0.0 blob_<number> gets removed:
```
Comment (snej) [inline C/tests/c4DocumentTest.cc]: This should be a new test, not a replacement of the existing one.

File: REST/RESTListener+Handlers.cc [hunk 2]
```diff
@@ -287,8 +287,11 @@ namespace litecore { namespace REST {
 
         // Encode body as Fleece (and strip _id and _rev):
         alloc_slice encodedBody = c4doc_encodeStrippingOldMetaProperties(body,
-                                                                         c4db_getFLSharedKeys(db));
-
+                                                                         c4db_getFLSharedKeys(db),
+                                                                         outError);
+        if (!encodedBody && outError && outError->domain != 0 && outError->code != 0)
```
Comment (snej) [inline REST/RESTListener+Handlers.cc]: No need to check `outError`; if `encodedBody` is null, that means the call failed. (And even if the caller passed null for `outError`, this method should still return false.)

**Comment by pasin** [comment]
Thanks @snej  for the review. I have made changes accordingly.

**Comment by pasin** [comment]
I fixed an indentation error and rebased. It's ready for review.

**Pull request merged**